reconcile/aws_support_cases_sos.py

```diff
@@ -0,0 +1,57 @@
+import logging
+
+import utils.gql as gql
+
+from reconcile.queries import AWS_ACCOUNTS_QUERY
+from reconcile.queries import GITLAB_INSTANCES_QUERY
+from utils.aws_api import AWSApi
+from utils.gitlab_api import GitLabApi
+
+
+def get_deleted_keys(accounts):
+    return {account['name']: account['deleteKeys']
+            for account in accounts
+            if account['deleteKeys'] not in (None, [])}
+
+
+def get_keys_to_delete(aws_support_cases):
+    search_pattern = 'We have become aware that the AWS Access Key '
+    keys = []
+    for account, cases in aws_support_cases.items():
+        for case in cases:
+            comms = case['recentCommunications']['communications']
```

Comment:
can you leave a comment here with the expected content of the communication? if they change it in the future it would be nice to have it handy

Reply:
also, pehaps we could come up with a regexp that validates that the response has the format we are expecting, and handle the exception otherwise

Reply:
added a comment with the reference.
we are currently not using regex, but splitting by `search_pattern`.
from what i've seen until now, the `search_pattern` is consistent across AWS support cases.